src/main/java/org/folio/circulation/resources/AbstractClaimedReturnedResource.java

```diff
@@ -0,0 +1,55 @@
+package org.folio.circulation.resources;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.Note;
+import org.folio.circulation.domain.NoteLink;
+import org.folio.circulation.domain.NoteLinkType;
+import org.folio.circulation.domain.NoteType;
+import org.folio.circulation.domain.validation.GeneralNoteTypeValidator;
+import org.folio.circulation.infrastructure.storage.notes.NoteTypesRepository;
+import org.folio.circulation.infrastructure.storage.notes.NotesRepository;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.utils.CollectionUtil;
+
+import io.vertx.core.http.HttpClient;
+
+public abstract class AbstractClaimedReturnedResource extends Resource {
+  protected static final String NOTE_MESSAGE = "Claimed returned item marked lost";
+  protected static final String NOTE_DOMAIN  = "loans";
+
+  protected boolean isClaimedReturned;
+
+  public AbstractClaimedReturnedResource(HttpClient client) {
+    super(client);
+  }
+
+  protected  Result<Loan> setIsClaimedReturned(Result<Loan> loanResult) {
+    loanResult.map(loan -> isClaimedReturned = loan.isClaimedReturned());
+    return loanResult;
+  }
+
+  protected CompletableFuture<Result<Loan>> createNote(Clients clients, Loan loan, boolean isClaimedReturned) {
+    final NotesRepository notesRepo = new NotesRepository(clients);
+    final NoteTypesRepository noteTypesRepo = new NoteTypesRepository(clients);
+    if (isClaimedReturned) {
+      noteTypesRepo.findByName("General note")
+        .thenApply(GeneralNoteTypeValidator::refuseIfNoteTypeNotFound)
+        .thenApply(r -> r.map(CollectionUtil::firstOrNull))
+        .thenCompose(r -> r.after(noteType -> notesRepo.create(createNote(noteType, loan))));
+    }
+    return CompletableFuture.completedFuture(Result.succeeded(loan));
```

Comment:
Is this method intended to always return a future that is already completed, irrespective of whether a note is created or not?

Reply:
Yes. In the case of the DeclaredLostResource the status may not be claimed returned and then should be declared lost and complete successfully, but not add the note.

Reply:
That isn't quite what I meant, let me try to explain better.

The code within the `if` block does not have a return statement and the `return` statement is not in an else branch, meaning that this method _always_ returns an immediately completed future with a successful result irrespective of whether the note creation works or not, or how long it takes

Does that make sense?

Reply:
Separately, I don't think I follow why this code needs to check whether the item is `Claimed returned` or not?

Removing that check does not cause any tests to fail, if it is important, then there needs to be a test to demonstrate that behaviour.

Reply:
I understand now. 

Sounds like I need to add a test to demonstrate that the note is not created when the item has not been claimed returned before it is declared lost. 

Would it be acceptable to just return the block in the if statement? If I'm understanding correctly that would propagate any failures in the case where the item was claimed returned.

Reply:
> Sounds like I need to add a test to demonstrate that the note is not created when the item has not been claimed returned before it is declared lost.

Yes, that would be good. I found these requirements to be confusing and so have asked Emma a question on the JIRA issue.

> Would it be acceptable to just return the block in the if statement? If I'm understanding correctly that would propagate any failures in the case where the item was claimed returned.

Yes, that would be acceptable. It needs to be of the same type, which it isn't at the moment. It would likely need to change to something like:
```
  return noteTypesRepo.findByName("General note")
      .thenApply(GeneralNoteTypeValidator::refuseIfNoteTypeNotFound)
      .thenApply(r -> r.map(CollectionUtil::firstOrNull))
      .thenCompose(r -> r.after(noteType -> notesRepo.create(createNote(noteType, loan))))
      .thenCompose(r -> r.after(note -> completedFuture(succeeded(loan))));
```